scene/audio/audio_stream_player_internal.cpp

```diff
@@ -0,0 +1,253 @@
+			}
+			stream_playbacks.clear();
+		} break;
+
+		case Node::NOTIFICATION_PAUSED: {
+			if (!node->can_process()) {
+				// Node can't process so we start fading out to silence
+				set_stream_paused(true);
+			}
+		} break;
+
+		case Node::NOTIFICATION_UNPAUSED: {
+			set_stream_paused(false);
+		} break;
+	}
+}
+
+Ref<AudioStreamPlayback> AudioStreamPlayerInternal::play_basic() {
+	Ref<AudioStreamPlayback> stream_playback;
+	if (stream.is_null()) {
+		return stream_playback;
+	}
+	ERR_FAIL_COND_V_MSG(!node->is_inside_tree(), stream_playback, "Playback can only happen when a node is inside the scene tree");
+	if (stream->is_monophonic() && is_playing()) {
+		stop();
+	}
+	stream_playback = stream->instantiate_playback();
+	ERR_FAIL_COND_V_MSG(stream_playback.is_null(), stream_playback, "Failed to instantiate playback.");
+
+	stream_playbacks.push_back(stream_playback);
+	active.set();
+	_set_process(true);
+	return stream_playback;
+}
+
+void AudioStreamPlayerInternal::set_stream_paused(bool p_pause) {
+	// TODO this does not have perfect recall, fix that maybe? If there are zero playbacks registered with the AudioServer, this bool isn't persisted.
+	for (Ref<AudioStreamPlayback> &playback : stream_playbacks) {
+		AudioServer::get_singleton()->set_playback_paused(playback, p_pause);
+	}
+}
+
+bool AudioStreamPlayerInternal::get_stream_paused() const {
+	// There's currently no way to pause some playback streams but not others. Check the first and don't bother looking at the rest.
+	if (!stream_playbacks.is_empty()) {
+		return AudioServer::get_singleton()->is_playback_paused(stream_playbacks[0]);
+	}
+	return false;
+}
+
+void AudioStreamPlayerInternal::validate_property(PropertyInfo &p_property) const {
+	if (p_property.name == "bus") {
+		String options;
+		for (int i = 0; i < AudioServer::get_singleton()->get_bus_count(); i++) {
+			if (i > 0) {
+				options += ",";
+			}
+			String name = AudioServer::get_singleton()->get_bus_name(i);
+			options += name;
+		}
+
+		p_property.hint_string = options;
+	}
+}
+
+void AudioStreamPlayerInternal::set_stream(Ref<AudioStream> p_stream) {
+	stop();
+	stream = p_stream;
+}
+
+void AudioStreamPlayerInternal::seek(float p_seconds) {
+	if (is_playing()) {
+		stop();
+		play_callable.call(p_seconds);
+	}
+}
+
+void AudioStreamPlayerInternal::stop() {
+	for (Ref<AudioStreamPlayback> &playback : stream_playbacks) {
+		AudioServer::get_singleton()->stop_playback_stream(playback);
+	}
+	stream_playbacks.clear();
+	active.clear();
+	_set_process(false);
+}
+
+void AudioStreamPlayerInternal::set_node(Node *p_node, const Callable &p_play_callable, bool p_physical) {
+	node = p_node;
+	play_callable = p_play_callable;
+	physical = p_physical;
+
+	AudioServer::get_singleton()->connect("bus_layout_changed", callable_mp((Object *)node, &Object::notify_property_list_changed));
+	AudioServer::get_singleton()->connect("bus_renamed", callable_mp((Object *)this, &Object::notify_property_list_changed).unbind(3));
```

Comment:
I'm not sure about this, but shouldn't the second connect also be to `node`?

```suggestion
	AudioServer::get_singleton()->connect("bus_renamed", callable_mp((Object *)node, &Object::notify_property_list_changed).unbind(3));
```

Reply:
Yes, I already fixed it locally.